src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -68,13 +74,15 @@ void AnnotationManager::add(const AnnotationID& id, const FillAnnotation& annota
 Update AnnotationManager::update(const AnnotationID& id, const SymbolAnnotation& annotation, const uint8_t maxZoom) {
     Update result = Update::Nothing;
 
+    mutex.lock();
```

Comment:
Manual locking/unlocking will leave a stranded lock in the early exit case. (It's an error case, but still best not to corrupt the object internals.)